server/models/user.js

```diff
@@ -8,33 +8,33 @@ const userSchema = new Schema({
   username: {
     type: String,
     required: true,
-    index: { unique: true }
+    index: { unique: true },
   },
   name: {
     type: Schema.Types.ObjectId,
     ref: 'Name',
-    required: true
+    required: true,
   },
   email: { type: String, required: true },
   password: { type: String, required: true },
   phone: String,
   // Check below fields, if present, user is of that type
   _donor: {
     type: Schema.Types.ObjectId,
-    ref: 'Donor'
+    ref: 'Donor',
   },
   _family: {
     type: Schema.Types.ObjectId,
-    ref: 'Family'
+    ref: 'Family',
   },
   _organizer: {
     type: Schema.Types.ObjectId,
-    ref: 'Organizer'
-  }
+    ref: 'Organizer',
+  },
 });
 
 // Hash password before saving
-userSchema.pre('save', next => {
+userSchema.pre('save', function(next) {
```

Comment:
we losing the arrow functions? shouldn't this be `(next) =>`

Reply:
As per the official mongoose docs:
> Do not declare methods using ES6 arrow functions (=>). Arrow functions explicitly prevent binding this, so your method will not have access to the document and the above examples will not work.
[mongoose docs: methods](http://mongoosejs.com/docs/guide.html#methods)